js/stdlib/ts/ALGO.ts

```diff
@@ -1992,13 +2021,17 @@ export const connectAccount = async (networkAccount: NetworkAccount): Promise<Ac
             });
             const vres = await decode(vi, vvs, args);
             debug({vres});
-            return isSafe ? ['Some', vres] : vres;
+            const safeResult = ['Some', vres];
+            viewCacheSet(v, k, {lastCheckedTime: now, lastValue: safeResult});
```

Comment:
However, what you have is wrong... you should make a test case that shows it wrong, but let me tell you why. You are keyed on `${v}.${k}`, but the value you're caching is `${v}.${k}` applied to `args`... so you'll do something like if the view is `f = (x) => x + y` where `y` is in SVS, then `f(3)` returns `5` and then `f(4)` would return `5` rather than `6`.

Reply:
Ah, yes, I was only looking at constant function views when I was writing this...